Lib/ast.py

```diff
@@ -788,7 +788,11 @@ def get_raw_docstring(self, node):
             return node
 
     def get_type_comment(self, node):
-        comment = self._type_ignores.get(node.lineno) or node.type_comment
+        lineno = getattr(node, 'lineno', None)
+        if lineno is None:
+            return None
+
+        comment = self._type_ignores.get(lineno) or node.type_comment
```

Comment:
I'm not so sure about this. The crash before at least made the user call `fix_missing_locations` which solves the problem. Now you're suggesting to change the behavior in possibly invalid ways:
- if there *is* a `node.type_comment` set but line number isn't, now it will be silently not returned;
- if there *is* a matching type ignore in `self._type_ignore`, it won't be found because the location information is missing and no type comment will be returned.

Since the new behavior makes the code silently miss returning type comments without any indication as to why, the programmer might have trouble finding why this is.